tests/Filesystem/FilesystemTest.php

```diff
@@ -44,6 +44,10 @@ public function testPutStoresFiles()
 
     public function testReplaceStoresFiles()
     {
+        if (DIRECTORY_SEPARATOR == '\\') {
```

Comment:
This greatly weakens the test by skipping it on _all_ Windows versions for _all_ PHP versions, instead of narrowing it down to Windows 10 + PHP 7.2.9 (according to your issue description itself). Btw a better check for OS would be the constant `PHP_OS` or `PHP_OS_FAMILY`, the latter is available starting from PHP 7.2 (which is just convenient).

Reply:
Nice, Closing for now, please submit any idea about improving it.

Reply:
> The main target environment is linux and it would not be affected or weakened by the change.

Notice that there are multiple assertions here, with only one or two related to symlinking. By skipping the test on every Windows version, with every PHP version, you're rendering the feature **totally** untested on Windows. Does it work? Does it not work? Will there be any BC break? No one will ever know anymore. This is what I mean by weakening the test, and is a big red flag.

Reply:
I do not believe in works in older versions.
And currently the error is causing all assertion to be skipped, anyway.

Maybe the test can be splited up into two tests to prevent the cascading the skippness or error to the rest of the assertions.